server/hotrod/src/main/scala/org/infinispan/server/hotrod/KeyValueVersionConverterFactory.scala

```diff
@@ -23,19 +22,39 @@ object KeyValueVersionConverterFactory {
          val capacity = UnsignedNumeric.sizeUnsignedInt(key.length) + key.length +
             (if (newValue != null) UnsignedNumeric.sizeUnsignedInt(newValue.length) + newValue.length + 8 else 0)
 
-         val out = ByteBuffer.allocate(capacity)
-         UnsignedNumeric.writeUnsignedInt(out, key.length)
-         out.put(key)
+         val out = Array.ofDim[Byte](capacity)
```

Comment:
Is this change really necessary? The overhead of ByteBuffer seems minimal to me, and o.i.commons.io.ByteBuffer has even less fields...